File: Backend/src/main/java/com/Trabajo_Final_Beltran/dto/response/NotificacionResponse.java
package com.Trabajo_Final_Beltran.dto.response;

import com.Trabajo_Final_Beltran.enums.TipoNotificacion;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NotificacionResponse {

  private Long id;

  private String titulo;

  private String mensaje;

  private Boolean leida;

  private LocalDateTime fecha;

  private TipoNotificacion tipo;

}

File: frontEnd/src/app/core/models/notificacion-response.ts
import { TipoNotificacion } from './enums/tipo-notificacion.enum';
import { TipoReferencia } from './enums/tipo-referencia.enum';

export interface NotificacionResponse {
  id: number;
  titulo: string;
  mensaje: string;
  leida: boolean;
  fecha: string;
  tipo: TipoNotificacion;
  tipoReferencia: TipoReferencia | null;
  referenciaId: number | null;
}
